src/input/input_handler.cpp

```diff
@@ -0,0 +1,694 @@
+    std::string token;
+
+    while (std::getline(ss, token, ',')) {
+        tokens.push_back(token);
+    }
+
+    // Check and process tokens
+    for (const auto& t : tokens) {
+        if (string_to_keyboard_key_map.find(t) != string_to_keyboard_key_map.end()) {
+            // Map to keyboard key
+            u32 key_id = string_to_keyboard_key_map.at(t);
+            if (!key1)
+                key1 = key_id;
+            else if (!key2)
+                key2 = key_id;
+            else if (!key3)
+                key3 = key_id;
+        } else if (string_to_axis_map.find(t) != string_to_axis_map.end()) {
+            // Map to axis input ID
+            u32 axis_id = GetAxisInputId(string_to_axis_map.at(t));
+            if (axis_id == (u32)-1) {
+                return InputBinding(0, 0, 0);
+            }
+            if (!key1)
+                key1 = axis_id;
+            else if (!key2)
+                key2 = axis_id;
+            else if (!key3)
+                key3 = axis_id;
+        } else if (string_to_cbutton_map.find(t) != string_to_cbutton_map.end()) {
+            // Map to controller button input ID
+            u32 cbutton_id = GetControllerButtonInputId((u32)string_to_cbutton_map.at(t));
+            if (cbutton_id == (u32)-1) {
+                return InputBinding(0, 0, 0);
+            }
+            if (!key1)
+                key1 = cbutton_id;
+            else if (!key2)
+                key2 = cbutton_id;
+            else if (!key3)
+                key3 = cbutton_id;
+        } else {
+            // Invalid token found; return default binding
+            return InputBinding(0, 0, 0);
+        }
+    }
+
+    return InputBinding(key1, key2, key3);
+}
+
+// function that takes a controlleroutput, and returns the array's corresponding element's pointer
+ControllerOutput* GetOutputPointer(const ControllerOutput& parsed) {
+    // i wonder how long until someone notices this or I get rid of it
+    int i = 0;
+    for (; i[output_array] != ControllerOutput(OrbisPadButtonDataOffset::None, Axis::AxisMax);
+         i++) {
+        if (i[output_array] == parsed) {
+            return &output_array[i];
+        }
+    }
+    return &output_array[i];
+}
+
+void ParseInputConfig(const std::string game_id = "") {
+
+    const auto config_file = Config::GetFoolproofKbmConfigFile(game_id);
+
+    // todo: change usages of this to GetFoolproofKbmConfigFile (in the gui)
+    if (game_id == "") {
+        return;
+    }
+
+    // we reset these here so in case the user fucks up or doesn't include this,
+    // we can fall back to default
+    connections.clear();
+    mouse_deadzone_offset = 0.5;
+    mouse_speed = 1;
+    mouse_speed_offset = 0.125;
+    int lineCount = 0;
+
+    std::ifstream file(config_file);
+    std::string line = "";
+    while (std::getline(file, line)) {
+        lineCount++;
+        // strip the ; and whitespace
+        line.erase(std::remove(line.begin(), line.end(), ' '), line.end());
```

Comment:
```suggestion
        line.erase(std::ranges::remove_if(line, std::isspace), line.end());
```
std::isspace will handle more cases

Reply:
```cpp
line.erase(std::remove_if(line.begin(), line.end(),
                          [](unsigned char c) { return std::isspace(c); }),
           line.end());
```
Your code was syntactically incorrect, but this should do the same